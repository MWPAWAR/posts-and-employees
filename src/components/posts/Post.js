import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { Link } from 'react-router-dom';
import { FormattedMessage } from 'react-intl';

const Post = ({
  id,
  body,
  title
}) => (
  <Card className="root">
    <CardActionArea>
      <CardContent>
        <Typography gutterBottom variant="h5" component="h2">
          {title}
        </Typography>
        <Typography variant="body1" color="textSecondary" component="p">
          {body}
        </Typography>
      </CardContent>
    </CardActionArea>
    <CardActions>
      <Link to={`/posts/${id}`}>
        <Button size="large" color="primary">
          <FormattedMessage id="app.posts.btn-view-label" />
        </Button>
      </Link>
    </CardActions>
  </Card>
);

Post.propTypes = {
  id: PropTypes.number.isRequired,
  body: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
};

export default Post;
