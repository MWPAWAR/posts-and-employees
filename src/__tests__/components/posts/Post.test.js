import React from 'react';
import { shallow } from 'enzyme';

import Post from '../../../components/posts/Post';
import postMock from '../../mocks/post.json'

describe('Post', () => {
  test('it renders correctly with default data', () => {
    const component = shallow(<Post {...postMock} />);
    expect(component).toMatchSnapshot();
  })
});
