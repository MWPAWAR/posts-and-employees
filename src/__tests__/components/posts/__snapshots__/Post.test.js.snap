// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post it renders correctly with default data 1`] = `
<WithStyles(ForwardRef(Card))
  className="root"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h2"
        gutterBottom={true}
        variant="h5"
      >
        Welcome to Posts
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body1"
      >
        Lorem Ipsum Lorem Ipsum Lorem Ipsum
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <Link
      to="/posts/1"
    >
      <WithStyles(ForwardRef(Button))
        color="primary"
        size="large"
      >
        <FormattedMessage
          id="app.posts.btn-view-label"
        />
      </WithStyles(ForwardRef(Button))>
    </Link>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
